components/style/stylesheets.rs

```diff
@@ -1836,9 +1854,10 @@ impl<'a, 'b> QualifiedRuleParser for NestedRuleParser<'a, 'b> {
     type QualifiedRule = CssRule;
 
     fn parse_prelude(&mut self, input: &mut Parser) -> Result<SelectorList<SelectorImpl>, ()> {
+        let ns = self.context.namespaces.expect("namespaces must be set when parsing rules").read();
```

Comment:
Yeah, this occurred to me, the solution is using something like `RefCell<&mut Namespaces>`

(`&&mut T` is equivalent to `&&T` semantics wise, so this works).

will file